### PR TITLE
Add poetry and bump dependencies

### PR DESCRIPTION
**Pull request by lykahb**
The configuration with pyproject.toml is a more declarative convention.

One breaking change is that the minimal version of Python is raised 3.7 to 3.9 to upgrade the dependencies to the most recent versions.

I verified that katrain works and ran the CI scripts manually for the test and release workflows. Testing the MacOSX packaging seems to be much easier with CI.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+readme = "README.md"
+packages = [{include = "katrain"}]
+include = [
+    {path = "katrain/KataGo/*"},
+    {path = "katrain/models/*"},
+    {path = "katrain/fonts/*"},
+    {path = "katrain/sounds/*"},
+    {path = "katrain/img/*"},
+    {path = "katrain/img/flags/*"},
+    {path = "katrain/i18n/*"}
```
Comment (lykahb) [inline pyproject.toml:21]: This logic from setup.py is more sophisticated, but the simpler configuration here seems to be equivalent.

```
package_data = {"": ["*.json", "*.kv", "*.wav"], "katrain": [], "tests": []}

def include_data_files(directory):
```

I ran the `python3 setup.py sdist` and `poetry build` with verbose mode and diffed their output to confirm that nothing extra is included.

File: .flake8 [hunk 2]
```diff
@@ -1,3 +1,4 @@
 [flake8]
-ignore = E501, E203, W503, E402 # line length, space before binary op, line break before binary op, import not at top
+# line length, space before binary op, line break before binary op, import not at top
+ignore = E501, E203, W503, E402
```
Comment (lykahb) [inline .flake8:3]: This config was invalid. Flake8 was failing with:
```
ValueError: Error code '#' supplied to 'ignore' option does not match '^[A-Z]{1,3}[0-9]{0,3}$'
```

File: pyproject.toml (hunk 1 above)
Comment (sanderland) [reply pyproject.toml:21]: can we remove it entirely and just rely on the default? I can't remember why this was added in the first place 🤔

**Comment by sanderland** [comment]
Can you also check the docs and see if they refer to requirements.txt?

File: pyproject.toml (hunk 1 above)
Comment (lykahb) [reply pyproject.toml]: Sure, I also verified that the same files get included.

**Comment by lykahb** [comment]
Rebased on the [sanderland:1.15.0](https://github.com/sanderland/katrain/tree/1.15.0) branch. There should be no more references to the older build system, in particular to the `requirements.txt`.

**Comment by sanderland** [comment]
Thank you! poetry-dynamic-versioning is a neat plugin as well.

**Comment by sanderland** [comment]
https://github.com/sanderland/katrain/actions/runs/10434894929/job/28898286719
Aside from pypi auth having changed, this doesn't look good:
> Publishing KaTrain (0.0.0) to PyPI

**Comment by lykahb** [comment]
Sorry about that, testing on a clean virtual env could've caught it

**Pull request merged**